icinga2/icinga.go

```diff
@@ -141,6 +157,25 @@ func (s *WebClient) FilteredQuery(url string, filter QueryFilter, result, errmsg
 	return s.napping.Send(&req)
 }
 
+func (s *WebClient) SetIcigaUrl(url string) {
+	s.URL = url
+}
+
+func (s *WebClient) TestIcingaApi(url string) error {
+	var results, errmsg Results
+
+	resp, err := s.napping.Get(url+"/v1", nil, &results, &errmsg)
+	if err != nil {
+		return err
+	}
+
+	if resp.HttpResponse().StatusCode != 200 {
```

Comment:
@pdorschner What if we get "301 Moved Permanently"?

Reply:
I see two cases here: either the HTTP client library we're using can follow redirects, so we should never get a "301 Moved Permanently" here, or it doesn't and this library won't be able to communicate with an Icinga2 instance which replies with "301 Moved Permanently" anyway.

However, I'd use `http.StatusOK` instead of the literal `200`:

```suggestion
	if resp.HttpResponse().StatusCode != http.StatusOK {
```